source/beamline/BioXAS/BioXASSSRLMonochromatorEnergyControl.cpp

```diff
@@ -194,6 +194,25 @@ AMAction3* BioXASSSRLMonochromatorEnergyControl::createCalibrateAction(double ol
 	return result;
 }
 
+double BioXASSSRLMonochromatorEnergyControl::calculateMoveProgressPercent() const
+{
+	double result = 0;
+
+	// The bragg motor is what's used to change the energy.
+	// We calculate the move progress based on the bragg motor positions.
+
+	if (bragg_) {
+		double braggMoveStart = bragg_->moveStart();
+		double braggMoveValue = bragg_->moveValue();
+		double braggMoveEnd = bragg_->moveEnd();
+
+		if (braggMoveStart != braggMoveEnd)
+			result = qAbs(braggMoveValue - braggMoveStart) / (braggMoveEnd - braggMoveStart);
```

Comment:
There is, as the bragg motor control inherits from AMControl. But this function is protected.